packages/schema/bind/src/bindings/rust/wasm/templates/entry-rs.mustache

```diff
@@ -12,20 +12,29 @@ use polywrap_wasm_rs::{
     invoke,
     InvokeArgs,
 };
+use crate::module::Module;
 
 #[no_mangle]
 pub extern "C" fn _wrap_invoke(method_size: u32, args_size: u32, env_size: u32) -> bool {
     // Ensure the abort handler is properly setup
     abort::wrap_abort_setup();
 
+    let mut module = Module::__new__();
     let args: InvokeArgs = invoke::wrap_invoke_args(method_size, args_size);
 
     match args.method.as_str() {
         {{#moduleType}}
         {{#methods}}
-        "{{name}}" => invoke::wrap_invoke(args, env_size, Some({{#toLower}}{{name}}{{/toLower}}_wrapped)),
+        "{{name}}" => {
+            let result = {{#toLower}}{{name}}{{/toLower}}_wrapped(&mut module, args.args.as_slice(), env_size);
```

Comment:
I think this part can be pulled out from the conditions so it's not repeated in each one: 

```rust
invoke::wrap_invoke_result(result);
true
```